app/assets/javascripts/discourse/app/services/presence.js

```diff
@@ -524,11 +524,19 @@ export default class PresenceService extends Service {
         ) {
           presentChannels.push(channelName);
         } else {
+          // The user is not present. Update the proxies and add the channel to the leave list.
+          Array.from(proxies).forEach((p) => {
+            this._removePresent(p);
+          });
```

Comment:
This change meant that users would never go back 'online' after the tab became active again. I added a test for this case, and reworked things so that we keep proper track of which channels are currently 'present but inactive'